google/cloud/spanner_v1/snapshot.py

```diff
@@ -129,7 +142,14 @@ def _restart_on_unavailable(
                 if transaction is not None:
                     transaction_selector = transaction._make_txn_selector()
                 request.transaction = transaction_selector
-                iterator = method(request=request)
+                nth_request = next_nth_request()
+                attempt = 1
```

Comment:
This should not reset the attempt to 1. This should increase the attempt, as this is a retry of the query due to an UNAVAILABLE error.

Reply:
Gotcha and thanks! Done.